backend/src/types/params.ts

```diff
@@ -34,7 +34,7 @@ export interface AddServiceArgs {
   service: {
     serviceName: string
     username: string
-    email: string
+    email: string | null
     reposurl: string
   }
```

Comment:
This type could be service: 
```
export interface AddServiceArgs {
  service: ServiceUserType & { id: string }
}
```
where ServiceUserType is imported from types/user. doing this would reduce repetitiveness and keep our types more tied together

Reply:
Did this and also simplified `authorizeWithGHMutationResult` in frontend with type intersection (didn't even know that was a thing, seems pretty useful)

Reply:
Also [Omit](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys) and [Pick are very useful](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys)